Tests/resources/directory/subdirectory/SubDirTestController.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Test\Subdirectory;
+
+class SubDirTestController
+{
+    public function init()
+    {
+        $this->l('SubdirShop');
+    }
+
+    public function productAction()
+    {
+        $this->trans('SubdirFingers', [], 'admin.product.help');
```

Comment:
Best follow good practices, even if it would work at extractor level
```suggestion
        $this->trans('SubdirFingers', [], 'Admin.Product.Help');
```